src/main/java/Objective.java

```diff
@@ -0,0 +1,23 @@
+import java.util.*;
+
+public class Objective {
```

Comment:
It seems to me more of the functionality of this class should be implemented inside the class. It only has two accessor methods, the goal and result members are never used.